lib/solveEquation/stepThrough.js

```diff
@@ -71,19 +71,19 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
 
     if (steps.length > 0) {
       const lastStep = steps[steps.length - 1];
-      equation = Equation.createEquationFromString(
-        lastStep.newEquation.ascii(), equation.comparator);
+      equation = lastStep.newEquation.clone();
```

Comment:
can you explain this change?

Reply:
Ahh good call--I should have left a comment explaining this change..

So, what this code did/does is create a new step using the previous step's finalEquation as its starting point, which makes sense.

However, what was odd about the way they did it was by first _printing_ the equation and then re-parsing it, which is odd considering that there is an equation.clone method that seems a lot more direct and avoids the hassle of printing and reparsing.

Why do this matter? The answer is that some information could get lost in in between printing and parsing, in particular with negative fraction coefficients as is described above with the parsing of `-⅓x` gets parsed as having x in the denominator. Eg you could start with a properly wrapped fraction like `-(1/3)x`, but then after removing unnecessary parens that would get turned into 

```
    *
   /  \
  -    x
  |
  ÷
 / \
 1  3
```

which is and accurate node tree, but the issue is that that tree prints to a string as  `-1/3x` which, when parsed back into a tree, will get parsed has having the x in the denominator, which is bad.

The simplest fix was just to use the actual equation tree rather than print and reparse and hope that everything stays the same

Reply:
cool makes sense. Might be worth asking the mathsteps author why she did it this way originally? Was it just a simple oversight that she didn't realize `.clone` is both easier and less error-prone?

Reply:
> cool makes sense. Might be worth asking the mathsteps author why she did it this way originally? Was it just a simple oversight that she didn't realize `.clone` is both easier and less error-prone?

I would ask, but I'm afraid that she doesn't want to be bothered